src/directory-sync/utils/get-primary-email.ts

```diff
@@ -0,0 +1,6 @@
+import { UserWithGroups } from '../interfaces/user.interface';
+
+export function getPrimaryEmail(user: UserWithGroups): string | undefined {
+  const primaryEmail = (user.emails || []).find((email) => email.primary);
+  return primaryEmail && primaryEmail.value;
```

Comment:
We should be able to use optional chaining here instead:

```suggestion
  return primaryEmail?.value;
```